### ISSUE TITLE
SWS for Warping Older Tracks for DJ Sets?

### ISSUE THREAD

**Issue by buckman1974** [reaper-oss/sws#1504]
I have asked this on the Reaper Forum, but I am looking for a quicker way to warp full songs that drift from the 70's 80s etc and I think Reaper 'could' do this better AND faster than Ableton live?

There maybe a particluar SWS Extension or combination Macro that could do this?

I import a 'wobbly' drifting track into Reapers timeline and cut it with Tab to Transient on the first downbeat.I add a stretch marker here (on the first beat) and then use Dynamic Split to keep the item intact but to place warp markers. These are often quite erratic as Reaper doesn't pick out the start of bars in the 70's 80s track.

It seems very close to Reaper being able to do this, as I use X-Raym-Add stretch Markers to selected items at grid divisions.eel But this just uses Reapers grid divisions (in order to timestretch and change tempo) but ignores the drifting track
If Reaper could almost analyze the 'track' and place stretch markers on the downbeats of the 70's 80s track, and then Quantize the Stretch Markers to Grid, would I believe achieve this and it would tighten up the track plus allow any tempo to be used as the item would be locked and synced.
Does anyone know of anything I am missing here, thats since been updated or extra extensions that would help this workflow?
I do use Ableton for this kinda thing which most says is best, but Ableton still requires a lot of tweaking once you start, and Reaper improves every week and I'm sure its stretching is much better to my ears.

Any Actions, Macros or video tutorials etc welcome

**Comment by vanhaze2000** [comment]
Great One, very much wanted indeed !!

If Reaper / SWS could have the ability to just recognize (ONLY) the kickdrums in an audiofile and place a Stretchmarker on each kickdrum, then that would be amazing.
**I am not a programmer but i assume some algorithm looking for both low frequency content (30-150 hz for example) AND an attack phase (Transient) would be necessary, to distinguish kickdrums in an audiofile.**

**Comment by buckman1974** [comment]
Do you think its possible then?

**Comment by vanhaze2000** [comment]
Yes, i think so;
The created Stretchmarkers on only kickdrums in an audiofile (old disco track) could then be quantized to Reaper 1/4 Grid.
Resulting in a perfect sync of the disco track to Reaper's Tempo Grid.

**_Ofcourse there could be an issue of artifacts in the sound, due to the stretching.
Important IMO is that the kick transients are "hurt the least" in the stretches, so they don't become sounding muffled._**

**Comment by buckman1974** [comment]
Yes you are right, i have tried many ways with actions, extra extensions, and others programming actions and macros and its just out of reach.
Unless you think its possible?
I have gone back to Serato Studio that does quantizing of sloppy 70's tracks and does the key!

I wish Reaper did this, as there is so much that Reaper does better as a full DAW

**Comment by vanhaze2000** [comment]
Maybe there does exist a script that can create Stretchmarkers perfectly at only kickdrums in a song, dunno really.

Me 2 using Serato Studio for that work, quite revolutionary DAW but has so many left to be desired.

You could also use **Serato Sample plugin** in Reaper (not a free plugin !), to load a sloppy discotrack into.
Serato Sample does the quantize job pretty well.
You could then ofcourse render the audio out from Sample offline (faster than online !), resulting in an Audiotrack containing the quantized song,
and then removing Sample plugin, thus less clutter in your Reaper Project and saving CPU resources.
I assume you could even write a Custom Action for the rendering > removing Sample plugin, for faster workflow.

**Comment by buckman1974** [comment]
Yes i have Serato Sample but MIDI triggering samples or full song sections did sound like a nightmare rather than using audio directly on the timeline.
But as you say i could render it out to audio but it just seemed a long way around it, when Reaper could probably do something similar or better inside the actual DAW


**_I assume you could even write a Custom Action for the rendering > removing Sample plugin, for faster workflow._**
This sounds good? so a single button push could turn to audio on the grid, and remove the plug in completely etc?

**Comment by vanhaze2000** [comment]
Yep !
- Assuming you have placed Serato Sample plugin in first FX slot of a SELECTED Track, the Custom Action would be:
Item: Apply track/take FX to items (multichannel output)
Script: me2beats_Remove focused track fx.lua

_For the last action to be available in your Reaper Action list, you must have ReaPack installed and
the me2beats repository URL imported into ReaPack, which is this URL:
https://github.com/me2beats/reapack/raw/master/index.xml_
